src/commands/diff.ts

```diff
@@ -0,0 +1,139 @@
+    {
+      name: 'old',
+      description: 'old spec path, URL or context-name',
+      required: true,
+    },
+    {
+      name: 'new',
+      description: 'new spec path, URL or context-name',
+      required: true,
+    },
+  ];
+
+  async run() {
+    const { args, flags } = this.parse(Diff); // NOSONAR
+    const firstDocumentPath = args['old'];
+    const secondDocumentPath = args['new'];
+
+    const outputFormat = flags['format'];
+    const outputType = flags['type'];
+    const overrideFilePath = flags['overrides'];
+
+    let firstDocument: Specification, secondDocument: Specification;
+
+    try {
+      firstDocument = await load(firstDocumentPath);
+    } catch (err) {
+      if (err instanceof SpecificationFileNotFound) {
+        this.error(
+          new ValidationError({
+            type: 'invalid-file',
+            filepath: firstDocumentPath,
+          })
+        );
+      } else {
+        this.error(err as Error);
+      }
+    }
+
+    try {
+      secondDocument = await load(secondDocumentPath);
+    } catch (err) {
+      if (err instanceof SpecificationFileNotFound) {
+        this.error(
+          new ValidationError({
+            type: 'invalid-file',
+            filepath: secondDocumentPath,
+          })
+        );
+      } else {
+        this.error(err as Error);
+      }
+    }
+
+    let overrides = {};
+    if (overrideFilePath) {
+      overrides = await readOverrideFile(overrideFilePath);
+    }
+
+    try {
+      const firstDocumentParsed = await parser.parse(firstDocument.text());
+      const secondDocumentParsed = await parser.parse(secondDocument.text());
+      const diffOutput = diff.diff(
+        firstDocumentParsed.json(),
+        secondDocumentParsed.json(),
+        {
+          override: overrides,
+        }
+      );
+
+      if (outputFormat === 'json') {
+        this.outputJson(diffOutput, outputType);
+      } else {
+        this.log(
+          `The output format ${outputFormat} is not supported at the moment.`
+        );
+      }
+    } catch (error) {
+      throw new ValidationError({
+        type: 'parser-error',
+        err: error,
+      });
+    }
+  }
+
+  outputJson(diffOutput: AsyncAPIDiff, outputType: string) {
+    if (outputType === 'breaking') {
+      this.log(JSON.stringify(diffOutput.breaking(), null, 2));
+    } else if (outputType === 'non-breaking') {
+      this.log(JSON.stringify(diffOutput.nonBreaking(), null, 2));
+    } else if (outputType === 'unclassified') {
+      this.log(JSON.stringify(diffOutput.unclassified(), null, 2));
+    } else if (outputType === 'all') {
+      this.log(JSON.stringify(diffOutput.getOutput(), null, 2));
+    } else {
+      this.log(`The output type ${outputType} is not supported at the moment.`);
+    }
+  }
+}
+
+async function readOverrideFile(path: string): Promise<diff.OverrideObject> {
+  const overrideStringData = await readFile(path, { encoding: 'utf8' });
+  return JSON.parse(overrideStringData);
+}
```

Comment:
Not sure where to put this function.

Reply:
imho it is fine if it just stays here, just add js doc

most important is to have proper error handling, awaits should be wrapped in try/catch and we need test, what error goes out when user passed path to not existing file and what error goes out when user passed path to not json file and `JSON.parse` fails